frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx

```diff
@@ -246,13 +258,95 @@ export default memo(function HeatMap({
       expandedTissues,
       setExpandedTissues,
       sortedCellTypesByTissueName,
+      value,
     ]
   );
 
+  const uniqueCellTypes = useMemo(() => {
+    const result: Set<string> = new Set<string>();
+    Object.values(sortedCellTypesByTissueName).forEach((cellTypes) => {
+      cellTypes.forEach((cellType) => {
+        result.add(cellType.cellTypeName);
+      });
+    });
+    return [...result].sort();
+  }, [sortedCellTypesByTissueName]);
+
```

Comment:
Adding a comment for clarity

```suggestion

// update displayedCellTypes and expandedTissues
```